### ISSUE TITLE
Improve renewing the ccache with LDAP when using GSSAPI

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#1586]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/1582
- Created at 2012-10-16 18:07:16 by [jhrozek](https://pagure.io/user/jhrozek)
- Closed at 2020-03-24 14:20:48 as wontfix
- Assigned to nobody
---

When the SSSD is configured to use GSSAPI, then we only kinit very close to the end of life of the ccache (lifetime - ldap_opt_timeout) and only once. We should change it so that we try at, say, 66% of the ccache lifetime and then retry again.

    17:07 <@jhrozek> I would propose to renew at 50%, if the first renewal fails, then try again at 75% and keep trying at reasonable intervals until we either fail or renew
    17:07 <@jhrozek> simo: also factor in logic not to renew to fast
    17:07 * JrAquino has seen several ccache errors in the sssd logs as of late
    17:08 < simo> jhrozek: I would say renew at 2/3 (66%) then retry backing off after 5 sec, 10 sec, 20 sec ... etc until you try once every 5 min.
    17:08 < simo> backing off is quite standard in this type of stuff

### Comments

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-10-18 17:20:07](https://pagure.io/SSSD/sssd/issue/1582#234404)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.10 beta
priority: major => minor

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-10-25 15:55:40](https://pagure.io/SSSD/sssd/issue/1582#234405)**

Fields changed

rhbz:  => todo

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2012-12-20 23:33:42](https://pagure.io/SSSD/sssd/issue/1582#234406)**

Fields changed

selected:  => Not need

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2013-01-02 15:32:09](https://pagure.io/SSSD/sssd/issue/1582#234407)**

Moving tickets that are not a priority for SSSD 1.10 into the next release.

milestone: SSSD 1.10 beta => SSSD 1.11 beta

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-09-26 18:49:57](https://pagure.io/SSSD/sssd/issue/1582#234408)**

Fields changed

mark:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2015-02-10 16:54:58](https://pagure.io/SSSD/sssd/issue/1582#234409)**

Fields changed

changelog:  => 
design:  => 
design_review:  => 0
fedora_test_page:  => 
milestone: SSSD 1.13 beta => SSSD 1.13 backlog
priority: minor => major
review:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2015-02-12 20:12:51](https://pagure.io/SSSD/sssd/issue/1582#234410)**

Mass-moving tickets not planned for the 1.13 release to 1.14

milestone: SSSD 1.13 backlog => SSSD 1.14 beta

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-02-16 12:04:58](https://pagure.io/SSSD/sssd/issue/1582#234411)**

Out of scope for 1.14, sorry.

milestone: SSSD 1.14 beta => SSSD 1.15 beta
sensitive:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-25 10:03:27](https://pagure.io/SSSD/sssd/issue/1582#234412)**

This ticket was not touched or needed for 4 years, but I still think it's a good idea, therefore I propose we move it to the 'patches welcome' milestone instead.

review: 0 => 1
selected: Not need => May

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-12-05 15:42:09](https://pagure.io/SSSD/sssd/issue/1582#234413)**

The idea still makes sense, but is not planned (and IMO wouldn't be planned w/o a patch), so I'm moving it to Patches welcome.

milestone: SSSD Future releases (no date set yet) => SSSD Patches welcome

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-02-24 14:49:38](https://pagure.io/SSSD/sssd/issue/1582#234414)**

**Metadata Update from @jhrozek**:
- Issue set to the milestone: SSSD Patches welcome

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-24 14:20:47](https://pagure.io/SSSD/sssd/issue/1582#636937)**

Thank you for taking time to submit this request for SSSD. Unfortunately this issue was not given priority and the team lacks the capacity to work on it at this time.

Given that we are unable to fulfill this request I am closing the issue as wontfix.

If the issue still persist on recent SSSD you can request re-consideration of this decision by reopening this issue. Please provide additional technical details about its importance to you.

Thank you for understanding.

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-24 14:20:48](https://pagure.io/SSSD/sssd/issue/1582#636938)**

**Metadata Update from @pbrezina**:
- Issue close_status updated to: wontfix
- Issue status updated to: Closed (was: Open)
